tests/e2e/multi_vc_vsphere.go

```diff
@@ -0,0 +1,416 @@
+	connectMultiVC(ctx, vs)
+	dcList, err := vs.getAllDatacentersForMultiVC(ctx)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	var vmMoRefForvmUUID object.Reference
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
+		var vmMoRefFound bool
+		for _, dc := range dcList {
+			for i := 0; i < len(vs.multiVcClient); i++ {
+				soapClient := vs.multiVcClient[i].Client.Client
+				vimClient, err := vim25.NewClient(ctx, soapClient)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				datacenter := object.NewDatacenter(vimClient, dc.Reference())
+				s := object.NewSearchIndex(vimClient)
+				vmUUID = strings.ToLower(strings.TrimSpace(vmUUID))
+				instanceUUID := !(vanillaCluster || guestCluster)
+				vmMoRef, err := s.FindByUuid(ctx, datacenter, vmUUID, true, &instanceUUID)
+
+				if err != nil || vmMoRef == nil {
+					continue
+				}
+				if vmMoRef != nil {
+					vmMoRefFound = true
+					vmMoRefForvmUUID = vmMoRef
+				}
+			}
+		}
+
+		if vmMoRefFound {
+			framework.Logf("vmuuid: %s still exists", vmMoRefForvmUUID)
+			continue
+		} else {
+			return nil, fmt.Errorf("node VM with UUID:%s is not found", vmUUID)
+		}
+	}
+	return vmMoRefForvmUUID, nil
+}
+
+/*
+verifyVolumeIsAttachedToVMInMultiVC checks volume is attached to the VM by vmUUID on a
+multivc environment. This function returns true if volume is attached to the VM, else returns
+false
+*/
+// func (vs *multiVCvSphere) verifyVolumeIsAttachedToVMInMultiVC(client clientset.Interface, volumeID string, vmUUID string) (bool, error) {
+// 	ctx, cancel := context.WithCancel(context.Background())
+// 	defer cancel()
+// 	vmRef, err := vs.getVMByUUIDForMultiVC(ctx, vmUUID)
+// 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+// 	framework.Logf("vmRef: %v for the VM uuid: %s", vmRef, vmUUID)
+// 	gomega.Expect(vmRef).NotTo(gomega.BeNil(), "vmRef should not be nil")
+
+// 	for i := 0; i < len(vs.multiVcClient); i++ {
+// 		soapClient := vs.multiVcClient[i].Client.Client
+// 		vimClient, err := vim25.NewClient(ctx, soapClient)
+// 		if err != nil {
+// 			return false, err
+// 		}
+// 		vm := object.NewVirtualMachine(vimClient, vmRef.Reference())
+// 		device, err := getVirtualDeviceByDiskID(ctx, vm, volumeID)
+// 		if err == nil && device == nil {
+// 			continue
+// 		}
+// 		if device != nil && err != nil {
+// 			framework.Logf("failed to determine whether disk %q is still attached to the VM with UUID: %q", volumeID, vmUUID)
+// 			continue
+// 		}
+// 		if device != nil && err == nil {
+// 			framework.Logf("Found the disk %q is attached to the VM with UUID: %q", volumeID, vmUUID)
+// 			return true, nil
+// 		}
+
+// 		if device == nil && err != nil {
+// 			continue
+// 		}
+// 	}
+
+// 	return false, nil
+// }
+
+func (vs *multiVCvSphere) verifyVolumeIsAttachedToVMInMultiVC(client clientset.Interface, volumeID string, vmUUID string) (bool, error) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Add appropriate details about what the method does

Reply:
done